src/main/java/org/baeldung/spring/CaptchaConfig.java

```diff
@@ -0,0 +1,22 @@
+package org.baeldung.spring;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.springframework.http.client.SimpleClientHttpRequestFactory;
+
+@Configuration
+@ComponentScan(basePackages = { "org.baeldung.captcha" })
+@ConditionalOnProperty(prefix = "google.recaptcha.key", name = { "site", "secret" })
+public class CaptchaConfig {
+    @Bean
+    public ClientHttpRequestFactory clientHttpRequestFactory() {
```

Comment:
Hey @pliesveld - let's keep the discussion on JIRA - as it gets hard to track everything if it's not in a single place. 
So - go ahead and copy-paste your questions over there. 
Cheers, 
Eugen.